RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+# RFC: Amplify JS TypeScript Improvements
+
+Amplify JS is looking to improve our TypeScript support across the library to better serve our customers and provide a more intuitive & idiomatic developer experience. To this end, we are requesting your feedback on a variety of changes and improvements that will be available to you in our next major version release.
+
+This RFC is broken down into sections covering:
+
+- Library-wide TypeScript improvements
+- Changes to some of our core utilities such as `Hub` and Amplify configuration
+- Specific improvements to the `Auth`, `Storage`, and `API` categories and associated APIs
+
+We're also requesting feedback on any other TypeScript issues or pain points that you may have encountered not explicitly covered in this RFC.
+
+# Library-wide TypeScript Improvements
+
+Amplify JS will be making the following improvements to our TypeScript support. These improvements will be applied across the entire library, not just the categories highlighted below.
+
+- `strict` typings — We will be applying `strict` mode to the entire library to improve the usability of our types.
```

Comment:
Will `strict` be enforced throughout or just on the public facing api surface? Think we're talking throughout in which case, making our usability better is only one of the benefits where maybe internal quality and consistency (or something like this) is improved by end to end enforcement.

Reply:
It's supposed to be enforced throughout the library, thus reflected on the public facing APIs as well.

Reply:
Agreed. Just think this is underselling the benefits